### ISSUE TITLE
Where does normalization of the output quaternion happen?

### ISSUE THREAD

**Issue by alykhantejani** [alexgkendall/caffe-posenet#6]
Hi @alexgkendall 

I've been looking through the `train_posenet.prototxt` and I can't see where the normalization of the predicted quaternion (from `cls3_fc_pose_wpqr`) happens. I see that the output of `cls3_fc_pose_wpqr` and the label go into the Euclidean loss (with weight `beta`), but in the paper equation `2` shows that the output quaternion is made a unit vector. 

Am I missing something here?

Thanks,
Aly

**Comment by alexgkendall** [comment]
Hey Aly,

That's correct - this functionality is not implemented in this Caffe code. This is something which I have in my Torch implementation of PoseNet. In practice, I have found it not to make a significant difference in the network's performance so I have omitted it here. Sorry for the confusion. 

You can see in the evaluation scripts that I normalise the resulting quaternion vector for testing.

**Comment by alykhantejani** [comment]
Hi @alexgkendall,

Thanks for the clarification. Is your torch implementation public yet (with/without pre-trained models)? 

Cheers,
Aly

**Comment by alexgkendall** [comment]
No, not at this stage.

**Comment by eeshandhekane** [comment]
Hi @alexgkendall !

Is the Torch implementation of PoseNet public yet?
